pybamm/citations.py

```diff
@@ -230,37 +226,41 @@ def print(self, filename=None, output_format="text", verbose=False):
             # delete the invalid citation from the set
             self._unknown_citations.remove(key)
 
-        if output_format == "text":
-            citations = pybtex.format_from_strings(
-                self._cited, style="plain", output_backend="plaintext"
-            )
-        elif output_format == "bibtex":
-            citations = "\n".join(self._cited)
-        else:
-            raise pybamm.OptionError(
-                f"Output format {output_format} not recognised."
-                "It should be 'text' or 'bibtex'."
-            )
+        citations = self._cited
 
         if filename is None:
-            print(citations)
+            for entry in citations:
+                print(self._string_formatting(entry))
             if verbose:
                 self._tag_citations()  # pragma: no cover
         else:
             with open(filename, "w") as f:
-                f.write(citations)
+                for entry in citations:
+                    f.write(self._string_formatting(entry))
+
+    def _string_formatting(self, entry):
+        bibtexparser = have_optional_dependency("bibtexparser")
+        if not isinstance(entry, bibtexparser.model.Entry):
+            raise TypeError()
```

Comment:
Coverage is failing as this is not covered by tests, would be nice to add some test case covering this part too.

Reply:
I agree, and a helpful error message would be nice as well – a `TypeError` in itself is not descriptive enough